plugins/logger/zap_adapter.go

```diff
@@ -17,20 +18,39 @@ func NewZapAdapter(zapLogger *zap.Logger) *ZapAdapter {
 	}
 }
 
+func separateFields(keyVals []interface{}) (fields []zap.Field, individualKeyVals []interface{}) {
+	for _, value := range keyVals {
+		switch value := value.(type) {
+		case zap.Field:
+			fields = append(fields, value)
+		case core.ObjectMarshaler:
+			fields = append(fields, zap.Inline(value))
```

Comment:
Could you please write a test case for the `core.ObjectMarshaler` ?

Reply:
I've added a test case for this, I'm not use it's the most efficient way of testing the log output to use a file but I couldn't work out how any output was tested from the other test cases. If you have a better way of doing this I'd be happy to update the PR.

There was a bug with the documentation with how the object is assigned. This is fixed and will now produce the correct inline output.

In all fairness the use of Object Marshalling could be removed completely and just have the end user provide a zap.Inline field instead.